### PR TITLE
Add an anchor field to the link plugin to handle anchors in URLs.

### PR DESCRIPTION
**Pull request by yrucrem**
runbuild

### CODE REVIEW

**Comment by GenticsDev** [comment]
Can one of the admins verify this patch?

File: src/plugins/common/link/lib/link-plugin.js [hunk 1]
```diff
@@ -706,6 +715,57 @@ define([
...
+				},
+
+				setInputValue: function(anchor) {
+					this.element.val(anchor);
+
+					if (anchor && (!this.isVisible() || !that.toggleAnchor.getState())) {
+						this.show();
+						that.toggleAnchor.setState(true);
+					} else if (!anchor && (this.isVisible() || that.toggleAnchor.getState())) {
+						this.hide();
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: Set focus to hrefField

File: src/plugins/common/link/lib/link-plugin.js [hunk 2]
```diff
@@ -706,6 +715,57 @@ define([
...
+					} else {
+						that.anchorField.show();
+						that.anchorField.focus();
+					}
+
+					return false;
+				}
+			});
+
+			this.anchorField = Ui.adopt('editAnchor', Text, {
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: Introduce configuration option to enable anchor field support (defaults to false)

File: src/plugins/common/link/lib/link-plugin.js [hunk 3]
```diff
@@ -706,6 +715,57 @@ define([
...
+				// on the "change" event (e.g. when the element loses
+				// focus). updateValue() is intended to be called for
+				// every keystroke.
+				updateValue: function(anchor) {
+					var hrefItem = that.hrefField.getItem();
+					var href = hrefItem
+						? hrefItem.url ? hrefItem.url : ''
+						: that.hrefField.getValue();
+
+					that.lastActiveLink.setAttribute(
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: Check for that.lastActiveLink !== false

File: src/plugins/common/link/lib/link-plugin.js [hunk 4]
```diff
@@ -717,17 +777,138 @@ define([
...
+				anchor = '';
+			} else {
+				anchor = href.substring(idx + 1);
+			}
+
+			this.anchorField.setInputValue(anchor);
+		},
+
+		/**
+		 * Remote the anchor from the href field, when a
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: Remote -> Remove

File: src/plugins/common/link/lib/link-plugin.js [hunk 5]
```diff
@@ -1134,11 +1327,15 @@ define([
...
 					}
-					
+
 					// register a timeout that will set the value as soon as the href field was initialized
 					that.hrefUpdateInt = setInterval( function () {
 						if (jQuery( '#' + that.hrefField.getInputId()).length > 0) { // the object was finally created
+							that.prepareAnchor(foundMarkup.href);
 							that.hrefField.setTargetObject(foundMarkup, 'href');
+							that.stripAnchor();
+							that.hrefChange();
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: addAnchor when hrefField.getItem() (like above)

File: src/plugins/common/ui/lib/text.js [hunk 6]
```diff
@@ -16,17 +16,42 @@ function (jQuery, Component) {
...
+		 * Note that this method is called on the "change" event,
+		 * which is not triggered on every keypress, but only
+		 * when the element loses focus or enter is pressed.
+		 *
+		 * @param {string} value The new value of the text input.
 		 */
-		setValue: function (value) {}
+		setValue: function (value) {},
+
+		getValue: function () {
```
Comment (npomaroli) [inline src/plugins/common/ui/lib/text.js]: Comment

**Comment by npomaroli** [comment]
runbuild

File: src/plugins/common/link/lib/link-plugin.js [hunk 7]
```diff
@@ -340,7 +348,10 @@ define([
 			if (typeof this.settings.hrefValue != 'undefined') {
 				this.hrefValue = this.settings.hrefValue;
 			}
-			
+			if (typeof this.settings.anchorLinks != 'undefined') {
+				this.anchorLinks = this.settings.anchorLinks && this.settings.anchorLinks !== 'false';
```
Comment (npomaroli) [inline src/plugins/common/link/lib/link-plugin.js]: check case insensitive

**Pull request merged**